src/btree/bt_random.c

```diff
@@ -408,13 +409,15 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags, WT_
         descent = NULL;
         for (i = 0; i < entries; ++i) {
             descent = pindex->index[__wt_random(rnd) % entries];
-            if (descent->state == WT_REF_DISK || descent->state == WT_REF_MEM)
+            descent_state = __wt_ref_get_state(descent);
```

Comment:
I figure as we're doing a volatile atomic load this should be fine.

Reply:
I'm mostly agreed. Atomic volatile reads have the same effect as READ_ONCE, and while I'd prefer READ_ONCE for clarity I think making it work with ref->__state is going to have a few annoying implementation details. Happy to keep this as is.